layers/+email/mu4e/packages.el

```diff
@@ -122,7 +122,27 @@
                 (push (buffer-name buffer) buffers))))
           (nreverse buffers)))
       (setq gnus-dired-mail-mode 'mu4e-user-agent)
-      (add-hook 'dired-mode-hook 'turn-on-gnus-dired-mode))))
+      (add-hook 'dired-mode-hook 'turn-on-gnus-dired-mode)))
+
+  (if (configuration-layer/package-usedp 'org)
```

Comment:
This must be refactored, you cannot simply inline multiple use-package calls. Instead these must be declared on the top-level with their respective condition in their `:if` construct or refactored to do their respective require steps in the `pre-init-org`.

Your test against the mu4e version is ok and should not require additional use-package declarations.

Reply:
I used `use-package` instead of `(autoload 'org-mu4e-compose-org-mode ...)` or `(with-eval-after-load 'org ...)` because I thought it is a nicer wrapper that achieves the same functionality. 

I am pretty new to both emacs and spacemacs. Could you kindly tell me if there's something extra that `use-package` does which is undesired here, or this is more of a convention rule within spacemacs?

Reply:
The big difference is that use-package implies a package installation, this however is controlled on the layer's top level by the mechanism of init-package-name. When you call use-package in an inner condition you are actually overwritting the layer system i.e. if someone did exclude the package of `org-mu4e` in his dotfile the layer would not trigger the package installation, in your case use-package would simply auto-install it if the right variable would be set as part of the layers loading process.